nvtabular/ops/operator.py

```diff
@@ -84,7 +84,72 @@ def compute_output_schema(self, input_schema: Schema, col_selector: ColumnSelect
             else output_selector
         )
 
-        return Schema(output_names)
+        # group (old, new) names, grab old if in new columns
+        names_original, names_transformed = [], []
+        for original_column in col_selector.names:
+            # logic works for all ops except TE
+            new_cols = [new_column for new_column in output_names if original_column in new_column]
+            if new_cols:
+                for column in new_cols:
+                    if column not in names_transformed:
+                        names_original.append(original_column)
+                        names_transformed.append(column)
+        # if name is same from input schema means it was a used column,
+        # in op logic to mutate against selector columns
+        new_names = [
+            name
+            for name in output_names
+            if name not in names_transformed and name not in input_schema.column_schemas
+        ]
+        new_col_schemas = []
+        # grab old column schemas and change the names
+        for orig_col_name, new_column_name in zip(names_original, names_transformed):
+            orig_col_schema = input_schema.column_schemas[orig_col_name]
+            new_col_schemas.append(orig_col_schema.with_name(new_column_name))
+        # create column Schemas for new columns
+        for new_col_name in new_names:
+            if new_col_name not in input_schema.column_schemas:
+                # needs to be filled in
+                new_col_schemas.append(ColumnSchema(new_col_name))
+        # add in all tags properties and update dtypes
+        final_new_cols = []
+        for column in new_col_schemas:
+            column = self._add_tags(column)
+            # column = self._add_properties(column)
+            column = self._update_dtype(column)
+            final_new_cols.append(column)
+        return Schema(final_new_cols)
+
+    def _add_tags(self, column_schema):
+        return column_schema.with_tags(self._get_tags())
+
+    def _add_properties(self, column_schema):
+        # get_properties should return the additional properties
+        # for target column
+        target_column_properties = self._get_properties()[column_schema.name]
+        return column_schema.with_properties(target_column_properties)
+
+    def _update_dtype(self, column_schema):
+        return column_schema.with_dtype(self._get_dtype())
+
+    def _get_dtype(self):
```

Comment:
Since each op has to override these, it seems like the name should make sense from the perspective of the op, so that the sub-class definitions of these functions read cleanly and make sense in context. Maybe `output_dtype`, `output_tags`, and `output_properties`?